lib/Monads/No_Fail.thy

```diff
@@ -17,58 +17,15 @@ begin
 subsection "Non-Failure"
 
 text \<open>
-  With the failure flag, we can formulate non-failure separately
-  from validity. A monad @{text m} does not fail under precondition
-  @{text P}, if for no start state in that precondition it sets
-  the failure flag.
+  With the failure flag, we can formulate non-failure separately from validity.
+  A monad @{text m} does not fail under precondition @{text P}, if for no start
+  state in that precondition it sets the failure flag.
```

Comment:
This is splitting hairs, but is it slightly better to say "no start state satisfying the precondition", rather than "in that precondition", seeing that these preconditions aren't actually sets?